app/src/main/java/com/cmput301w23t40/capturetheqr/LibraryActivity.java

```diff
@@ -70,11 +87,42 @@ public boolean onMove(@NonNull RecyclerView cityList, @NonNull RecyclerView.View
             @Override
             public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
                 // TODO: delete via Firestore (so changes are persistent)
-                qrCodeDataList.remove(viewHolder.getBindingAdapterPosition());
-                qrCodeList.notifyDataSetChanged();
 
+                //qrCodeDataList.remove(viewHolder.getBindingAdapterPosition());
+
+                if (qrCodeDataList.size() > 0) {
+                    DB.delQRCodeInDB(qrCodeDataList.get(viewHolder.getBindingAdapterPosition()).getHashValue());
+
+                    qrCodeList.notifyDataSetChanged();
+
+                }
             }
         }).attachToRecyclerView(qrCodeView);
+
+        //real time updates from Firebase
+//        collectionRefQR=DB.getCollectionReferenceQR();
+        DB.getCollectionReferenceQR().addSnapshotListener(new EventListener<QuerySnapshot>() {
+            @Override
+            public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable
+            FirebaseFirestoreException error) {
+                // Clear the old list
+                //qrCodeDataList.clear();
+                for(QueryDocumentSnapshot doc: queryDocumentSnapshots)
+                {
+                    Log.d(TAG, String.valueOf(doc.getData().get("BFG5DGW54")));
+                    String qrcode = doc.getId();
+                    String qrid = (String) doc.getData().get("BFG5DGW54");
+                    qrCodeDataList.remove(new QRCode(qrcode, qrid,visFake, 10)); // Adding the cities and provinces from FireStore
```

Comment:
I am not sure why you want to locally delete the entries you just retrived from the DB?  Shouldn't it be that you first clear the list you have locally, and then you retrieve the entries from the DB and fill them in the local list, finally you notify the changes?